### PR TITLE
c4core: add version 0.1.10 and adapot conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **c4core/***

- migrate conan v2
- add version 0.1.10
- add with_fast_float option
- update dependencies

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

File: recipes/c4core/all/conandata.yml [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+patches:
+  "1.0.10":
+    - patch_file: "0.1.10-0001-make-fast_float-external.patch"
+      patch_description: "use cci's fast_float recipe"
+  "1.0.9":
+    - patch_file: "0.1.8-0001-make-fast_float-external.patch"
+      patch_description: "use cci's fast_float recipe"
+  "1.0.8":
+    - patch_file: "0.1.8-0001-make-fast_float-external.patch"
+      patch_description: "use cci's fast_float recipe"
```
Comment (prince-chrismc) [inline recipes/c4core/all/conandata.yml:21]:
```suggestion
  "1.0.10":
    - patch_file: "0.1.10-0001-make-fast_float-external.patch"
      patch_description: "use cci's fast_float recipe"
      patch_type: "conan"
  "1.0.9":
    - patch_file: "0.1.8-0001-make-fast_float-external.patch"
      patch_description: "use cci's fast_float recipe"
      patch_type: "conan"
  "1.0.8":
    - patch_file: "0.1.8-0001-make-fast_float-external.patch"
      patch_description: "use cci's fast_float recipe"
      patch_type: "conan"
```

File: recipes/c4core/all/conanfile.py [hunk 2]
```diff
@@ -1,8 +1,14 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get, rm, rmdir, replace_in_file, apply_conandata_patches
```
Comment (prince-chrismc) [inline recipes/c4core/all/conanfile.py:3]: Linter

```suggestion
from conan.tools.files import copy, get, rm, rmdir, apply_conandata_patches
```

File: recipes/c4core/all/conanfile.py [hunk 3]
```diff
@@ -42,49 +45,53 @@ def configure(self):
...
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["C4CORE_WITH_FASTFLOAT"] = bool(self.options.with_fast_float)
```
Comment (prince-chrismc) [inline recipes/c4core/all/conanfile.py:71]: cache variables should be used sparingly, those go at the command line which is great for override policy and such


```suggestion
        tc.variables["C4CORE_WITH_FASTFLOAT"] = bool(self.options.with_fast_float)
```

Comment (toge) [reply recipes/c4core/all/conanfile.py:71]: @prince-chrismc 
Thank you for your comment.
I now understand the difference between cache_variables and variables.

File: recipes/c4core/all/conanfile.py [hunk 4]
```diff
@@ -42,49 +45,56 @@ def configure(self):
...
-        ## clang with libc++ is not supported. It is already fixed at 2022-01-03.
-        if tools.Version(self.version) <= "0.1.8":
-            if (self.settings.compiler == "clang" and self.settings.compiler.get_safe("libcxx") == "libc++"):
-                raise ConanInvalidConfiguration(
-                    "{}/{} doesn't support clang with libc++".format(self.name, self.version),
-                )
+        ## clang with libc++ is not supported. It is already fixed since 0.1.9.
+        if Version(self.version) <= "0.1.8":
+            if self.settings.compiler in ["clang", "apple-clang"] and \
+                self.settings.compiler.get_safe("libcxx") == "libc++":
```
Comment (uilianries) [inline recipes/c4core/all/conanfile.py:62]:
```suggestion
            if self.info.settings.compiler in ["clang", "apple-clang"] and \
                self.info.settings.compiler.get_safe("libcxx") == "libc++":
```

File: recipes/c4core/all/conanfile.py [hunk 5]
```diff
@@ -42,49 +45,56 @@ def configure(self):
...
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        self.copy(pattern="LICENSE*", dst="licenses", src=self.source_folder)
```
Comment (uilianries) [inline recipes/c4core/all/conanfile.py:84]:
```suggestion
        copy(self, pattern="LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```

**Review by prince-chrismc — APPROVED** [review]

**Comment by uilianries** [comment]
@toge for some reason, Jenkins can not checkout this PR, and we had the same problem with another PR. I recommend you closing this PR and opening a new one, adding a new check (it can be anything, a empty line). It worked for us. Sorry for the inconvenient

**Comment by toge** [comment]
@uilianries 
I see.
I recreate PR.

**Pull request closed without merging**